pyiceberg/catalog/rest/__init__.py

```diff
@@ -678,7 +679,16 @@ def list_tables(self, namespace: Union[str, Identifier]) -> List[Identifier]:
 
     @retry(**_RETRY_ARGS)
     def load_table(self, identifier: Union[str, Identifier]) -> Table:
-        response = self._session.get(self.url(Endpoints.load_table, prefixed=True, **self._split_identifier_for_path(identifier)))
+        params = {}
+        if mode := self.properties.get(SNAPSHOT_LOADING_MODE):
```

Comment:
is it possible to add this as a parameter to load_table that takes preference over session-level properties?  I guess it's not part of the base catalog interface but 🤷🏻‍♂️

Reply:
Good seeing you here @corleyma, and I share your concern here. In some situations, we want to override this as well:

- To enable time-travel.
- In the case of a write, we want to know about all the snapshots to see if there are any conflicts.

I'm open to adding an option to the base interface, however, it does not make a lot of sense in the case of non-REST catalogs where we have to read the JSON anyway.